packages/back-end/src/models/BaseModel.ts

```diff
@@ -722,14 +755,15 @@ export abstract class BaseModel<
     // Resources with a single project
     if ("project" in obj && obj.project && typeof obj.project === "string") {
       const projects = await this.context.getProjects();
-      if (!projects.some((p) => p.id === obj.project)) {
+      if (!projects || !projects.some((p) => p.id === obj.project)) {
         throw new Error("Invalid project");
       }
     }
     // Resources with multiple projects
     else if ("projects" in obj && obj.projects && Array.isArray(obj.projects)) {
       const projects = await this.context.getProjects();
       if (
+        !projects ||
```

Comment:
So, I'm still a little fuzzy on it, but, my IDE was throwing an error here that `projects` could be null. [Link to error](https://github.com/growthbook/growthbook/actions/runs/9748833851/job/26904678074)

I think it may be because of how we're instantiating `rawDocs` as an undefined value within `_find()` and then later reassigning it, either via the config.yml file or the Mongo results.

If I update the code to something like:

```ts
  protected async _find(
    query: FilterQuery<Omit<z.infer<T>, "organization">> = {},
    {
      sort,
      limit,
      skip,
      bypassReadPermissionChecks,
    }: {
      sort?: Partial<
        {
          [key in keyof Omit<z.infer<T>, "organization">]: 1 | -1;
        }
      >;
      limit?: number;
      skip?: number;
      bypassReadPermissionChecks?: boolean;
    } = {}
  ) {
    const queryWithOrg = {
      ...query,
      organization: this.context.org.id,
    };

    if (this.useConfigFile()) {
      const docs =
        this.getConfigDocuments().filter((doc) =>
          evalCondition(doc, queryWithOrg)
        ) || [];

      if (!docs.length) return [];

      sort &&
        docs.sort((a, b) => {
          for (const key in sort) {
            const typedKey = key as keyof z.infer<T>;
            const sortDir = sort[typedKey] as 1 | -1;

            if (a[typedKey] < b[typedKey]) return -1 * sortDir;
            if (a[typedKey] > b[typedKey]) return 1 * sortDir;
          }
          return 0;
        });

      const migrated = docs.map((d) =>
        this.migrate(this._removeMongooseFields(d))
      );
      const filtered = bypassReadPermissionChecks
        ? migrated
        : await this.filterByReadPermissions(migrated);

      if (!skip && !limit) return filtered;

      return filtered.slice(skip || 0, limit ? (skip || 0) + limit : undefined);
    } else {
      const cursor = this._dangerousGetCollection().find(queryWithOrg);
      sort &&
        cursor.sort(
          sort as {
            [key: string]: 1 | -1;
          }
        );
      const docs = await cursor.toArray();

      if (!docs.length) return [];

      const migrated = docs.map((d) =>
        this.migrate(this._removeMongooseFields(d))
      );
      const filtered = bypassReadPermissionChecks
        ? migrated
        : await this.filterByReadPermissions(migrated);

      if (!skip && !limit) return filtered;

      return filtered.slice(skip || 0, limit ? (skip || 0) + limit : undefined);
    }
  }
  ```
  
  The error goes away.
  
  I haven't found the exact smoking gun yet as to why, reading through the code, it looks like we'd never return null, and the generated signature doesn't appear to change.